// echarts-theme-rsuite/dist/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target, mod));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  dark: () => rsuite_dark_default,
  light: () => rsuite_light_default
});
module.exports = __toCommonJS(src_exports);

// src/rsuite_light.ts
var import_lodash = __toESM(require("lodash.merge"));
var theme = {
  color: [
    "#34c3ff",
    "#a873e6",
    "#13ba9e",
    "#ee5765",
    "#f5a623",
    "#2575fc",
    "#df6ecd",
    "#8338ec",
    "#e6b980",
    "#51E8FF"
  ],
  backgroundColor: "rgba(0,0,0,0)",
  textStyle: {},
  title: {
    textStyle: {
      color: "#333333"
    },
    subtextStyle: {
      color: "#aaaaaa"
    }
  },
  line: {
    itemStyle: {
      borderWidth: "0"
    },
    lineStyle: {
      width: 2
    },
    symbolSize: "0",
    symbol: "circle",
    smooth: false
  },
  radar: {
    itemStyle: {
      borderWidth: "0"
    },
    lineStyle: {
      width: 2
    },
    symbolSize: "0",
    symbol: "circle",
    smooth: false
  },
  bar: {
    itemStyle: {
      barBorderWidth: 0,
      barBorderColor: "#ccc"
    }
  },
  pie: {
    itemStyle: {
      borderWidth: 0,
      borderColor: "#ccc"
    }
  },
  scatter: {
    itemStyle: {
      borderWidth: 0,
      borderColor: "#ccc"
    }
  },
  boxplot: {
    itemStyle: {
      borderWidth: 0,
      borderColor: "#ccc"
    }
  },
  parallel: {
    itemStyle: {
      borderWidth: 0,
      borderColor: "#ccc"
    }
  },
  sankey: {
    itemStyle: {
      borderWidth: 0,
      borderColor: "#ccc"
    }
  },
  funnel: {
    itemStyle: {
      borderWidth: 0,
      borderColor: "#ccc"
    }
  },
  gauge: {
    itemStyle: {
      borderWidth: 0,
      borderColor: "#ccc"
    }
  },
  candlestick: {
    itemStyle: {
      color: "#e01f54",
      color0: "#001852",
      borderColor: "#f5e8c8",
      borderColor0: "#b8d2c7",
      borderWidth: 1
    }
  },
  graph: {
    itemStyle: {
      borderWidth: 0,
      borderColor: "#ccc"
    },
    lineStyle: {
      width: 1,
      color: "#aaaaaa"
    },
    symbolSize: "0",
    symbol: "circle",
    smooth: false,
    color: [
      "#34c3ff",
      "#a873e6",
      "#13ba9e",
      "#ee5765",
      "#f5a623",
      "#2575fc",
      "#df6ecd",
      "#8338ec",
      "#e6b980",
      "#51E8FF"
    ],
    label: {
      color: "#eeeeee"
    }
  },
  map: {
    itemStyle: {
      areaColor: "#eeeeee",
      borderColor: "#444444",
      borderWidth: 0.5
    },
    label: {
      color: "#000000"
    },
    emphasis: {
      itemStyle: {
        areaColor: "rgba(255,215,0,0.8)",
        borderColor: "#444444",
        borderWidth: 1
      },
      label: {
        color: "rgb(100,0,0)"
      }
    }
  },
  geo: {
    itemStyle: {
      areaColor: "#eeeeee",
      borderColor: "#444444",
      borderWidth: 0.5
    },
    label: {
      color: "#000000"
    },
    emphasis: {
      itemStyle: {
        areaColor: "rgba(255,215,0,0.8)",
        borderColor: "#444444",
        borderWidth: 1
      },
      label: {
        color: "rgb(100,0,0)"
      }
    }
  },
  categoryAxis: {
    axisLine: {
      show: true,
      lineStyle: {
        color: "#e5e5ea"
      }
    },
    axisTick: {
      show: false,
      lineStyle: {
        color: "#333"
      }
    },
    axisLabel: {
      show: true,
      color: "#575757"
    },
    splitLine: {
      show: false,
      lineStyle: {
        color: ["#e5e5ea"]
      }
    },
    splitArea: {
      show: false,
      areaStyle: {
        color: ["rgba(250,250,250,0.3)", "rgba(200,200,200,0.3)"]
      }
    }
  },
  valueAxis: {
    axisLine: {
      show: false,
      lineStyle: {
        color: "#e5e5ea"
      }
    },
    axisTick: {
      show: false,
      lineStyle: {
        color: "#8e8e93"
      }
    },
    axisLabel: {
      show: true,
      color: "#575757"
    },
    splitLine: {
      show: true,
      lineStyle: {
        color: ["#e5e5ea"]
      }
    },
    splitArea: {
      show: false,
      areaStyle: {
        color: ["rgba(250,250,250,0.3)", "rgba(200,200,200,0.3)"]
      }
    }
  },
  logAxis: {
    axisLine: {
      show: true,
      lineStyle: {
        color: "#e5e5ea"
      }
    },
    axisTick: {
      show: true,
      lineStyle: {
        color: "#333"
      }
    },
    axisLabel: {
      show: true,
      color: "#575757"
    },
    splitLine: {
      show: true,
      lineStyle: {
        color: ["#e5e5ea"]
      }
    },
    splitArea: {
      show: false,
      areaStyle: {
        color: ["rgba(250,250,250,0.3)", "rgba(200,200,200,0.3)"]
      }
    }
  },
  timeAxis: {
    axisLine: {
      show: true,
      lineStyle: {
        color: "#e5e5ea"
      }
    },
    axisTick: {
      show: true,
      lineStyle: {
        color: "#333"
      }
    },
    axisLabel: {
      show: true,
      color: "#575757"
    },
    splitLine: {
      show: true,
      lineStyle: {
        color: ["#e5e5ea"]
      }
    },
    splitArea: {
      show: false,
      areaStyle: {
        color: ["rgba(250,250,250,0.3)", "rgba(200,200,200,0.3)"]
      }
    }
  },
  toolbox: {
    iconStyle: {
      borderColor: "#999999"
    },
    emphasis: {
      iconStyle: {}
    }
  },
  legend: {
    textStyle: {
      color: "#8e8e93"
    }
  },
  tooltip: {
    axisPointer: {
      lineStyle: {
        color: "#cccccc",
        width: 1
      },
      crossStyle: {
        color: "#cccccc",
        width: 1
      }
    }
  },
  timeline: {
    lineStyle: {
      color: "#293c55",
      width: 1
    },
    itemStyle: {
      color: "#293c55",
      borderWidth: 1
    },
    controlStyle: {
      color: "#293c55",
      borderColor: "#293c55",
      borderWidth: 0.5
    },
    checkpointStyle: {
      color: "#e43c59",
      borderColor: "#c23531"
    },
    label: {
      color: "#293c55"
    },
    emphasis: {
      itemStyle: {
        color: "#a9334c"
      },
      controlStyle: {
        color: "#293c55",
        borderColor: "#293c55",
        borderWidth: 0.5
      },
      label: {
        color: "#293c55"
      }
    }
  },
  visualMap: {
    color: ["#c2edff", "#009de6"]
  },
  dataZoom: {
    backgroundColor: "rgba(47,69,84,0)",
    dataBackgroundColor: "rgba(47,69,84,0.3)",
    fillerColor: "rgba(167,183,204,0.4)",
    handleColor: "#a7b7cc",
    handleSize: "100%",
    textStyle: {
      color: "#333333"
    }
  },
  markPoint: {
    label: {
      color: "#eeeeee"
    },
    emphasis: {
      label: {
        color: "#eeeeee"
      }
    }
  }
};
var additionalSettings = {
  textStyle: {
    color: "#575757"
  },
  legend: {
    textStyle: {
      fontSize: 14
    }
  }
};
var rsuite_light_default = (0, import_lodash.default)(additionalSettings, theme);

// src/rsuite_dark.ts
var import_lodash2 = __toESM(require("lodash.merge"));
var theme2 = {
  color: [
    "#34c3ff",
    "#a873e6",
    "#13ba9e",
    "#ee5765",
    "#f5a623",
    "#2575fc",
    "#df6ecd",
    "#8338ec",
    "#e6b980",
    "#51E8FF"
  ],
  backgroundColor: "#1a1d24",
  textStyle: {},
  title: {
    textStyle: {
      color: "#ffffff"
    },
    subtextStyle: {
      color: "#aaaaaa"
    }
  },
  line: {
    itemStyle: {
      borderWidth: "0"
    },
    lineStyle: {
      width: 2
    },
    symbolSize: "0",
    symbol: "circle",
    smooth: false
  },
  radar: {
    itemStyle: {
      borderWidth: "0"
    },
    lineStyle: {
      width: 2
    },
    symbolSize: "0",
    symbol: "circle",
    smooth: false
  },
  bar: {
    itemStyle: {
      barBorderWidth: 0,
      barBorderColor: "#cccccc"
    }
  },
  pie: {
    itemStyle: {
      borderWidth: 0,
      borderColor: "#cccccc"
    }
  },
  scatter: {
    itemStyle: {
      borderWidth: 0,
      borderColor: "#cccccc"
    }
  },
  boxplot: {
    itemStyle: {
      borderWidth: 0,
      borderColor: "#cccccc"
    }
  },
  parallel: {
    itemStyle: {
      borderWidth: 0,
      borderColor: "#cccccc"
    }
  },
  sankey: {
    itemStyle: {
      borderWidth: 0,
      borderColor: "#cccccc"
    }
  },
  funnel: {
    itemStyle: {
      borderWidth: 0,
      borderColor: "#cccccc"
    }
  },
  gauge: {
    itemStyle: {
      borderWidth: 0,
      borderColor: "#cccccc"
    }
  },
  candlestick: {
    itemStyle: {
      color: "#e01f54",
      color0: "#001852",
      borderColor: "#f5e8c8",
      borderColor0: "#b8d2c7",
      borderWidth: 1
    }
  },
  graph: {
    itemStyle: {
      borderWidth: 0,
      borderColor: "#cccccc"
    },
    lineStyle: {
      width: 1,
      color: "#aaaaaa"
    },
    symbolSize: "0",
    symbol: "circle",
    smooth: false,
    color: [
      "#34c3ff",
      "#a873e6",
      "#13ba9e",
      "#ee5765",
      "#f5a623",
      "#2575fc",
      "#df6ecd",
      "#8338ec",
      "#e6b980",
      "#51E8FF"
    ],
    label: {
      color: "#e9ebf0"
    }
  },
  map: {
    itemStyle: {
      areaColor: "#eeeeee",
      borderColor: "#444444",
      borderWidth: 0.5
    },
    label: {
      color: "#000000"
    },
    emphasis: {
      itemStyle: {
        areaColor: "rgba(255,215,0,0.8)",
        borderColor: "#444444",
        borderWidth: 1
      },
      label: {
        color: "rgb(100,0,0)"
      }
    }
  },
  geo: {
    itemStyle: {
      areaColor: "#eeeeee",
      borderColor: "#444444",
      borderWidth: 0.5
    },
    label: {
      color: "#000000"
    },
    emphasis: {
      itemStyle: {
        areaColor: "rgba(255,215,0,0.8)",
        borderColor: "#444444",
        borderWidth: 1
      },
      label: {
        color: "rgb(100,0,0)"
      }
    }
  },
  categoryAxis: {
    axisLine: {
      show: true,
      lineStyle: {
        color: "#3c3f43"
      }
    },
    axisTick: {
      show: false,
      lineStyle: {
        color: "#333"
      }
    },
    axisLabel: {
      show: true,
      color: "#e9ebf0"
    },
    splitLine: {
      show: false,
      lineStyle: {
        color: ["#e5e5ea"]
      }
    },
    splitArea: {
      show: false,
      areaStyle: {
        color: ["rgba(250,250,250,0.3)", "rgba(200,200,200,0.3)"]
      }
    }
  },
  valueAxis: {
    axisLine: {
      show: false,
      lineStyle: {
        color: "#e5e5ea"
      }
    },
    axisTick: {
      show: false,
      lineStyle: {
        color: "#8e8e93"
      }
    },
    axisLabel: {
      show: true,
      color: "#e9ebf0"
    },
    splitLine: {
      show: true,
      lineStyle: {
        color: ["#3c3f43"]
      }
    },
    splitArea: {
      show: false,
      areaStyle: {
        color: ["rgba(250,250,250,0.3)", "rgba(200,200,200,0.3)"]
      }
    }
  },
  logAxis: {
    axisLine: {
      show: true,
      lineStyle: {
        color: "#3c3f43"
      }
    },
    axisTick: {
      show: true,
      lineStyle: {
        color: "#333"
      }
    },
    axisLabel: {
      show: true,
      color: "#e9ebf0"
    },
    splitLine: {
      show: true,
      lineStyle: {
        color: ["#3c3f43"]
      }
    },
    splitArea: {
      show: false,
      areaStyle: {
        color: ["rgba(250,250,250,0.3)", "rgba(200,200,200,0.3)"]
      }
    }
  },
  timeAxis: {
    axisLine: {
      show: true,
      lineStyle: {
        color: "#3c3f43"
      }
    },
    axisTick: {
      show: true,
      lineStyle: {
        color: "#333"
      }
    },
    axisLabel: {
      show: true,
      color: "#e9ebf0"
    },
    splitLine: {
      show: true,
      lineStyle: {
        color: ["#3c3f43"]
      }
    },
    splitArea: {
      show: false,
      areaStyle: {
        color: ["rgba(250,250,250,0.3)", "rgba(200,200,200,0.3)"]
      }
    }
  },
  toolbox: {
    iconStyle: {
      borderColor: "#999999"
    },
    emphasis: {
      iconStyle: {}
    }
  },
  legend: {
    textStyle: {
      color: "#8e8e93"
    }
  },
  tooltip: {
    axisPointer: {
      lineStyle: {
        color: "#cccccc",
        width: 1
      },
      crossStyle: {
        color: "#cccccc",
        width: 1
      }
    }
  },
  timeline: {
    lineStyle: {
      color: "#293c55",
      width: 1
    },
    itemStyle: {
      color: "#293c55",
      borderWidth: 1
    },
    controlStyle: {
      color: "#293c55",
      borderColor: "#293c55",
      borderWidth: 0.5
    },
    checkpointStyle: {
      color: "#e43c59",
      borderColor: "#c23531"
    },
    label: {
      color: "#293c55"
    },
    emphasis: {
      itemStyle: {
        color: "#a9334c"
      },
      controlStyle: {
        color: "#293c55",
        borderColor: "#293c55",
        borderWidth: 0.5
      },
      label: {
        color: "#293c55"
      }
    }
  },
  visualMap: {
    color: ["#c2edff", "#009de6"]
  },
  dataZoom: {
    backgroundColor: "rgba(47,69,84,0)",
    dataBackgroundColor: "rgba(47,69,84,0.3)",
    fillerColor: "rgba(167,183,204,0.4)",
    handleColor: "#a7b7cc",
    handleSize: "100%",
    textStyle: {
      color: "#333333"
    }
  },
  markPoint: {
    label: {
      color: "#e9ebf0"
    },
    emphasis: {
      label: {
        color: "#e9ebf0"
      }
    }
  }
};
var additionalSettings2 = {
  textStyle: {
    color: "#e9ebf0"
  },
  legend: {
    textStyle: {
      fontSize: 14
    }
  }
};
var rsuite_dark_default = (0, import_lodash2.default)(theme2, additionalSettings2);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  dark,
  light
});
